azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1256,76 +1361,287 @@ private void fillUpdateTimeAndExecId(List<ExecutableFlow> flows,
-      if (host == null) {
-        if (other.host != null)
-          return false;
-      } else if (!host.equals(other.host))
-        return false;
-      if (port != other.port)
-        return false;
-      return true;
+    public void run() {
+      while (!shutdown) {
+        synchronized (this) {
+          try {
+            long currentTime = System.currentTimeMillis();
+            if (isActive
+              && currentTime - QUEUE_PROCESSOR_WAIT_IN_MS > lastProcessingTime) {
+              // Refresh executor stats to be used by selector
+              refreshExecutors();
+              // process upto a maximum of maxContinousSubmission from queued
+              // flows
+              processQueuedFlows(maxContinousSubmission);
+              lastProcessingTime = currentTime;
+            }
+            wait(QUEUE_PROCESSOR_WAIT_IN_MS);
+          } catch (InterruptedException e) {
+            logger.info("Interrupted. Probably to shut down.");
+          }
+        }
+      }
     }
   }
 
-  @Override
-  public int getExecutableFlows(int projectId, String flowId, int from,
-      int length, List<ExecutableFlow> outputList)
-      throws ExecutorManagerException {
-    List<ExecutableFlow> flows =
-        executorLoader.fetchFlowHistory(projectId, flowId, from, length);
-    outputList.addAll(flows);
-    return executorLoader.fetchNumExecutableFlows(projectId, flowId);
+  private void refreshExecutors() {
+    synchronized (activeExecutors) {
+      // TODO: rest api call to refresh executor stats
+    }
   }
 
-  @Override
-  public List<ExecutableFlow> getExecutableFlows(int projectId, String flowId,
-      int from, int length, Status status) throws ExecutorManagerException {
-    return executorLoader.fetchFlowHistory(projectId, flowId, from, length,
-        status);
+  private void processQueuedFlows(long maxContinousSubmission) {
+    try {
+      Pair<ExecutionReference, ExecutableFlow> runningCandidate;
+      int submissionNum = 0;
+      while (submissionNum < maxContinousSubmission
+        && (runningCandidate = queuedFlows.peek()) != null) {
+        synchronized (runningCandidate) {
+
+          ExecutionReference reference = runningCandidate.getFirst();
+          ExecutableFlow exflow = runningCandidate.getSecond();
+
+          // TODO: use dispatcher
+          Executor choosenExecutor;
+
+          synchronized (activeExecutors) {
+            choosenExecutor = activeExecutors.iterator().next();
+          }
+
+          if (choosenExecutor != null) {
+            queuedFlows.poll();
+            queuedFlowMap.remove(exflow.getExecutionId());
+
+            try {
+              reference.setExecutor(choosenExecutor);
+              executorLoader.assignExecutor(choosenExecutor.getId(),
+                exflow.getExecutionId());
+              // TODO: ADD rest call to do an actual dispatch
+              callExecutorServer(reference, ConnectorParams.EXECUTE_ACTION);
+              runningFlows
```

Comment:
where is the logic to remove the flow from the queuedFlow list after a successful dispatch ?